### DISCUSSION TITLE
How to set fields on Model in OnModelBeforeCreate Event Hook

### DISCUSSION THREAD

**Discussion by DigNZ** [pocketbase/pocketbase#2414] [Q&A] [answered]
Maybe its just been a long day.

I can use OnRecordBeforeCreateRequest to get a record and can Set() a value on it and that works. But this hook is NOT triggered if the record is created via models.NewRecord  ... / SaveRecord methods (or the form.Submit version). 

OnModelBeforeCreate IS triggered for both API and SaveRecord events but I can't work out how to set values on the record via e.ModelEvent.

I want the hook to trigger no matter how the record is created - what is the correct approach here?

**Comment by ganigeorgiev** [comment, marked as answer]
You can use type assertion on the `Model` interface, something like:

```go
app.OnModelBeforeCreate("yourCollection").Add(func(e *core.ModelEvent) error {
	rec := e.Model.(*models.Record)

	rec.Set("yourField", 123)

	return nil
})
```

**Reply by DigNZ** [reply]
Of course. Thanks so much.
